File: CPP/No191.cpp
/**
 * Created by Xiaozhong on 2020/8/20.
 * Copyright (c) 2020/8/20 Xiaozhong. All rights reserved.
 */
#include "iostream"

using namespace std;

class Solution {
public:
    int hammingWeight(uint32_t n) {
        int sum = 0;
        while (n != 0) {
            sum += (n & 1);
            n = n >> 1;
        }
        return sum;
    }
};

int main() {
    Solution s;
    cout << s.hammingWeight(uint32_t(11));
}

File: Java/src/No191.java
/*
 * Copyright (c) 2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

public class No191 {
    public int hammingWeight(int n) {
        int ans = 0, index = 0;
        while (index <= 31) {
            if ((n & (1 << index++)) != 0) ++ans;
        }
        return ans;
    }

    public static void main(String[] args) {
        System.out.println(new No191().hammingWeight(11));
    }
}
